Src/Newtonsoft.Json/Serialization/NamingStrategy.cs

```diff
@@ -101,5 +101,49 @@ public virtual string GetDictionaryKey(string key)
         /// <param name="name">The property name to resolve.</param>
         /// <returns>The resolved property name.</returns>
         protected abstract string ResolvePropertyName(string name);
+
+        /// <summary>
+        /// Hash code calculation
+        /// </summary>
+        /// <returns></returns>
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = GetType().GetHashCode();     // make sure different types do not result in equal values
+                hashCode = (hashCode * 397) ^ ProcessDictionaryKeys.GetHashCode();
+                hashCode = (hashCode * 397) ^ ProcessExtensionDataNames.GetHashCode();
+                hashCode = (hashCode * 397) ^ OverrideSpecifiedNames.GetHashCode();
+                return hashCode;
+            }
+        }
+
+        /// <summary>
+        /// Object equality implementation
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <returns></returns>
+        public override bool Equals(object obj)
+        {
+            if (!(obj is NamingStrategy other))
+            {
+                return false;
+            }
+
+            return Equals(other);
+        }
+
+        /// <summary>
+        /// Compare to another NamingStrategy
+        /// </summary>
+        /// <param name="other"></param>
+        /// <returns></returns>
+        protected bool Equals(NamingStrategy other)
+        {
+            return GetType() == other.GetType() &&
```

Comment:
`other` might be `null`. In `Equals(object)` I would just try to cast `obj as NamingStrategy` instead of using the `is` operator.

Reply:
The only call to this method is from public override bool Equals(object obj) which already does the null check.

Reply:
It's protected so it may be called from a derived class.

Reply:
Added null check.

Reply:
Since you do the null check here you can use the `as` operator instead in the other Equals method and make it a one-liner, perhaps as an expression-bodied member.

`public override bool Equals(object obj) => Equals(obj as NamingStrategy);`

Reply:
Thank you, made another fix :)